servlet.py

```diff
@@ -342,6 +350,45 @@ def get(self):
                 scaleMtLog(400, after-before, tInfo, key, len(toTranslate))
 
 
+class TranslatePageHandler(TranslateHandler):
+    @tornado.web.asynchronous
+    @gen.coroutine
+    def get(self):
+        logging.info(self.get_argument('langpair'))
+        try:
+            l1, l2 = self.get_argument('langpair').split('|')
+        except ValueError:
+            self.send_error(400, explanation='That pair is invalid, use e.g. eng|spa')
+        if '%s-%s' % (l1, l2) in self.pairs:
+#            mode = "%s-%s" % (l1, l2)
+            modeFile = self.pairs["%s-%s" % (l1, l2)]
+            modesdir = os.path.dirname(os.path.dirname(modeFile))
+            mode = os.path.splitext(os.path.basename(modeFile))[0]
+            url = self.get_argument('url')
+            data = urllib.request.urlopen(url).read()
+            if chardet:
+                encoding = chardet.detect(data)["encoding"]
+            else:
+                encoding = 'utf-8'
+            text = data.decode(encoding)   
+            text = text.replace('href="/',  'href="{uri.scheme}://{uri.netloc}/'.format(uri=urlparse(url)))
+            text = re.sub(r'a([^>]+)href=[\'"]?([^\'" >]+)', 'a \\1 href="#" onclick=\'window.parent.translateLink("\\2");\'', text)
+            logging.info(text)
+            commands = [['apertium', '-d', modesdir, '-f', "html", mode]]
+            logging.info("1")
+            result = yield translation.translateSimple(text, commands)
```

Comment:
look at what `get` in TranslateHandler does – this `get` should do the same, except for turning the url into a text first. In particular the actual call to translate needs to use `translated = yield pipeline.translate(toTranslate)`. 

We want to have caps on how many pipelines of a pair can be open at once, so the server admin doesn't have to worry about massive amounts of translation requests creating a new pair on every request. Imagine 100 people at the same time translate web sites; we don't want to start 100 pipelines at once and eat all our RAM (some of those analysers are huge). The `TranslateHandler.get` method avoids this by keeping at most N pipelines open at any one time.